src/App.tsx

```diff
@@ -15,36 +23,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function sortGoods(goods: string[], sortField: SortType, reversed: boolean) {
+  const sortedGoods = [...goods].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return reversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState(false);
+
+  const goods = sortGoods(goodsFromServer, sortField, reversed);
```

Comment:
The `goods` variable is correctly assigned the sorted goods. Ensure that `sortGoods` is called only when necessary to avoid unnecessary re-renders.